pystac/stac_object.py

```diff
@@ -346,7 +346,10 @@ def set_parent(self, parent: Optional[Catalog]) -> None:
             self.add_link(Link.parent(parent))
 
     def get_stac_objects(
-        self, rel: Union[str, pystac.RelType], typ: Optional[Type[STACObject]] = None
+        self,
+        rel: Union[str, pystac.RelType],
+        typ: Optional[Type[STACObject]] = None,
+        _prefer: Optional[str] = None,
```

Comment:
Yeah I didn't like that either, that's why I tried to hide it in the lowest function that seems like it wouldn't really be used externally :see_no_evil: but I like your approach way better.

Reply:
In changing this to `sort_function` I realized that really any link list alteration could be done with this callable because I will just be doing `links = sort_function(links)` if it is set. So someone could come in and use this callable to implement filtering or something. I am wondering if we should make the kwarg name more general to make that clear. Something like `preprocess_links`.

Alternatively we could limit the scope of this by only allowing the `sort_key` to be passed. This would be defined as a valid input for `sorted(list, key=<sort_key>)` which is any callable that takes a single input and returns a "sortable" key.

Reply:
> I am wondering if we should make the kwarg name more general to make that clear. Something like preprocess_links.

👍🏽 but suggest `modify_links` as the arg name?